kclvm/tools/src/LSP/src/find_refs.rs

```diff
@@ -0,0 +1,194 @@
+use crate::from_lsp::kcl_pos;
+use crate::goto_def::goto_definition;
+use crate::util::{build_word_index, parse_param_and_compile, Param};
+use anyhow;
+use kclvm_config::modfile::get_pkg_root;
+use lsp_types::Location;
+
+pub(crate) fn find_refs<F: Fn(String) -> Result<(), anyhow::Error>>(
+    def_loc: Location,
+    name: String,
+    cursor_path: String,
+    logger: F,
+) -> anyhow::Result<Option<Vec<Location>>> {
+    // todo: decide the scope by the workspace root and the kcl.mod both, use the narrower scope
+    // todo: should use the current file path
+    if let Some(root) = get_pkg_root(def_loc.uri.path()) {
+        match build_word_index(root) {
+            std::result::Result::Ok(word_index) => {
+                if let Some(locs) = word_index.get(name.as_str()).cloned() {
+                    return anyhow::Ok(Some(
+                        locs.into_iter()
+                            .filter(|ref_loc| {
+                                // from location to real def
+                                // return if the real def location matches the def_loc
+                                let file_path = ref_loc.uri.path().to_string();
+                                match parse_param_and_compile(
+                                    Param {
+                                        file: file_path.clone(),
+                                    },
+                                    None,
+                                ) {
+                                    Ok((prog, scope, _)) => {
+                                        let ref_pos = kcl_pos(&file_path, ref_loc.range.start);
+                                        // find def from the ref_pos
+                                        if let Some(real_def) =
```

Comment:
U can use the `cargo clippy` or the `cargo fix` command to simply all match statements.